ocs_ci/ocs/platform_nodes.py

```diff
@@ -220,6 +223,37 @@ def get_vms(self, nodes):
             vms.extend(node_vms)
         return vms
 
+    def get_vm_nodes_in_dc(self, nodes, dc=None):
+        """
+        Get vSphere vm objects list in the Datacenter(and not just in the cluster scope).
+        Note: If one of the nodes failed with an exception, it will not return his
+        corresponding VM object.
+
+        Args:
+            nodes (list): The OCS objects of the nodes
+            dc (str): The Datacenter name. If not specified, it will use the current dc.
+
+        Returns:
+            list: vSphere vm objects list in the Datacenter
+
+        """
+        dc = dc or self.datacenter
+        vms_in_dc = self.vsphere.get_all_vms_in_dc(dc)
+        node_names = set([node.get().get("metadata").get("name") for node in nodes])
+        vms = []
+        for vm in vms_in_dc:
+            try:
+                vm_name = vm.name
+                if vm_name in node_names:
+                    vms.append(vm)
+            except Exception as e:
+                logger.info(f"Failed to get the vm name due to exception: {e}")
+
+        if len(vms) < len(nodes):
+            logger.warning("Didn't find all the VM objects for all the nodes")
+
+        return vms
```

Comment:
this function will get all the VM's instance of Datacenter with matching node names of cluster and then its calling in https://github.com/red-hat-storage/ocs-ci/pull/5109/files#diff-478e5b871f475f1e312384c0eeb216726f11da23e3d3ac4df80aa3ee5d92e441R458

So basically this will delete all the the nodes for all the clusters in Datacenter

Reply:
If you looking for a VM instance of a particular cluster, you can use https://github.com/red-hat-storage/ocs-ci/blob/f2171eaf52a8fef11f3c523f3b427b7863cb5c9f/ocs_ci/ocs/platform_nodes.py#L203

Reply:
The function `get_vms` searches the node's objects in a specific pool, and not in all the Datacenter. And what we need right now with the IPI cluster is searching in all the Datacenter - that's why I created this new function.

Reply:
We will delete only the nodes specified in the `nodes` list.

Reply:
> 
If it is for IPI type, can we have it separate class instead of in common class ( VMWareNodes  )

Reply:
Yes, I will create a new class for IPI.